File: src/main/java/com/nanaios/polygonal_tech/client/PolygonalTechClient.java
package com.nanaios.polygonal_tech.client;

import com.nanaios.polygonal_tech.PolygonalTech;
import net.minecraft.client.renderer.texture.TextureAtlas;
import net.minecraftforge.client.event.TextureStitchEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent;

@Mod.EventBusSubscriber(modid = PolygonalTech.MODID, bus = Mod.EventBusSubscriber.Bus.MOD)
public class PolygonalTechClient {
    @SubscribeEvent
    public static void clientSetup(FMLClientSetupEvent event) {
        event.enqueueWork(PolygonalTechRenderType::setRenderTypes);
        event.enqueueWork(PolygonalTechScreen::registerScreens);
    }
}

File: src/main/kotlin/com/nanaios/polygonal_tech/main/client/PolygonalTechClient.kt
package com.nanaios.polygonal_tech.main.client

import com.nanaios.polygonal_tech.main.PolygonalTech
import net.minecraft.client.gui.screens.MenuScreens
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent

/**
 * クライアントサイドでのみ実行される画面の登録やレンダリング関連の設定を集約的に行うことを目的としたオブジェクト。
 */
@Mod.EventBusSubscriber(modid = PolygonalTech.MOD_ID,bus = Mod.EventBusSubscriber.Bus.MOD)
object PolygonalTechClient {
    /**
     * クライアントの初期化フェーズにて発火し、[MenuScreens]に対するGUI登録などをメインスレッドで安全にエンキューすることを目的とするメソッド。
     *
     * @param event クライアント用のセットアップイベント（[FMLClientSetupEvent]）
     */
    @JvmStatic
    @SubscribeEvent
    fun onClientSetup(event: FMLClientSetupEvent) {
        event.enqueueWork {
        }
    }
}